accounts/abi/unpack_test.go

```diff
@@ -943,3 +943,29 @@ func TestOOMMaliciousInput(t *testing.T) {
 		}
 	}
 }
+
+func TestPackAndUnpackIncompatibleNumber(t *testing.T) {
```

Comment:
Can you please extend this test a bit to include 

- valid inputs and the expected decoded value
- invalid inputs and the expected error

Reply:
> Can you please extend this test a bit to include
> 
> * valid inputs and the expected decoded value
> * invalid inputs and the expected error

Done.

Reply:
Done